servant-client/src/Servant/Client/Internal/HttpClient.hs

```diff
@@ -153,15 +158,18 @@ instance Alt ClientM where
   a <!> b = a `catchError` \_ -> b
 
 instance RunClient ClientM where
-  runRequestAcceptStatus = performRequest
+  runRequestAcceptStatus statuses req = do
+    ClientEnv _ _ _ _ mid <- ask
```

Comment:
suggestion: Use `NamedFieldPuns` to remove the need for positional pattern matching which will make code more robust in the face of change
suggestion: `mid` -> `middleware` will make it clearer what this function is about